helm/hpcc/templates/_helpers.tpl

```diff
@@ -811,12 +811,28 @@ Get image name
 {{- end -}}
 
 {{/*
-Generates imagePattern as env variable
+Generates image information into env. variables used at runtime for runtime platform version switching
 */}}
 {{- define "hpcc.generateImageEnv" -}}
 {{- /* Pass in a dictionary with root and me defined */ -}}
-- name: imagePattern
-  value: {{ include "hpcc.imageName" . }}
+{{- $baseImageRootName := "" -}}
+{{- $baseImageVersion := "" -}}
+{{- if .me.image -}}
+  {{- $baseImageRootName = printf "%s/%s" (.me.image.root | default .root.Values.global.image.root | default "hpccsystems") (.me.image.name | default .root.Values.global.image.name | default "platform-core") -}}
+{{- else -}}
+  {{- $baseImageRootName = printf "%s/%s" (.root.Values.global.image.root | default "hpccsystems") (.root.Values.global.image.name | default "platform-core") -}}
+{{- end -}}
+{{- if .me.image -}}
```

Comment:
trivial: could combine the two ifs  (on identical conditions)

Reply:
will change